### ISSUE TITLE
Seems like texture does not want to be set.

### ISSUE THREAD

**Issue by MayorZero** [rougier/freetype-gl#190]
I built ftgl and run 2 demos: ansi and font
It works perfect.

when I want to use it in my code there are problems.

https://github.com/MayorZero/Text-Renderer.git

debugger triggers error in glTexImage2D with GL_INVALID_VALUE type.
None of those errors should appear, because:
1. `is generated if width is less than 0 or greater than GL_MAX_TEXTURE_SIZE. ` - width is 512
2. `is generated if target is not GL_TEXTURE_1D_ARRAY or GL_PROXY_TEXTURE_1D_ARRAY and height is less than 0 or greater than GL_MAX_TEXTURE_SIZE.` - target is GL_TEXTURE_2D and height is 512
3. `is generated if target is GL_TEXTURE_1D_ARRAY or GL_PROXY_TEXTURE_1D_ARRAY and height is less than 0 or greater than GL_MAX_ARRAY_TEXTURE_LAYERS. ` - height is 512
4. `is generated if level is less than 0. ` - it is set to 0
5. `may be generated if level is greater than log2(max), where max is the returned value of GL_MAX_TEXTURE_SIZE` - level is 0
6. `is generated if internalFormat is not one of the accepted resolution and format symbolic constants.` GL_RED is a default one 
7. `is generated if width or height is less than 0 or greater than GL_MAX_TEXTURE_SIZE.` - 512x512 
8. `is generated if border is not 0. ` - it is set to 0

GL_MAX_TEXTURE_SIZE is greater than 512 for sure.

How to fix it?
Why does it not work?

**Comment by rougier** [comment]
I'm lost, what is the error code you get?
Probably a copy/paster mistake but you write `50"` instead of `50`

**Comment by DonTweaks** [comment]
You're generating mipmap once and never again after setData method call.

**Comment by MayorZero** [comment]
Yes, but in sparky engine it works perfectly fine. Check it out on git.

**Comment by DonTweaks** [comment]
Hmmmm. I see. I do not know what to say...

**Comment by MayorZero** [comment]
I would really like make it works, because it is a really nice system.

**Comment by DonTweaks** [comment]
Have you tried different, similar way to create a gl texture?

**Comment by MayorZero** [comment]
Yes, I have. I have generated a texture inside the `Font` class, added data to it and then set `m_texture` from `GLTexture` class to be this generated font texture (because of `SummitTexture` method in Renderer2D - it needs texture object, not id).
It seemed as if it is working, but text was without a texture, only black quads.

**Comment by DonTweaks** [comment]
At a first glance shaders look good. Have you tried this way with normal textures (player, map blocks, etc.)?

**Comment by MayorZero** [comment]
Yes. It renders without problems. I guess it is a ft or ftgl problem.

**Comment by MayorZero** [comment]
It is working. I want to use my renderer though, because of the performance. So... if freetype works, it must be a freetype gl problem.

**Comment by DonTweaks** [comment]
Or it may be texture class issue. I cannot help :/

**Comment by MayorZero** [comment]
@rougier  What do you think about it?

**Comment by DonTweaks** [comment]
In a first case you're using luminance alpha, that is why you're getting error. luminance and luminance alpha are replaced by GL_RED and GL_RG in open gl > version 3

**Comment by MayorZero** [comment]
You are right! Now I get no texture on, only black quads represented text.
